src/glmfit.jl

```diff
@@ -465,7 +469,7 @@ Fit a generalized linear model to data.
 $FIT_GLM_DOC
 """
 function fit(::Type{M},
-    X::Union{Matrix{T},SparseMatrixCSC{T}},
+    X::AbstractMatrix,
```

Comment:
below (line 496) we convert `X` with `float`, but here we do not do it. why?

Reply:
The method at line 496 supposed to promote the arguments before this one is called. However, I can see that this one should then continue be restricted floats. I'll fix that.